docs/slurm-perf-cluster.md

```diff
@@ -0,0 +1,182 @@
+SlurmPerf Deployment Guide
```

Comment:
We wanted to get rid of "SlurmPerf Deployment Guide." The top level name of this document should be "High-Performance Multi-Node Cluster Deployment Guide" to be more in-line with what the doc is aiming to acocmplish.

Reply:
Changed top-level header as suggested